src/main/java/dk/bec/polonez/reservationsystem/controller/OfferController.java

```diff
@@ -20,18 +20,32 @@ public OfferController(OfferService offerService) {
     }
 
     @GetMapping
-    public List<Offer> getAllOffers() {
+    public List<OfferDto> getAllOffers() {
         return offerService.getAll();
     }
 
     @GetMapping("{id}")
-    public Offer getOfferById(@PathVariable long id) {
+    public OfferDto getOfferById(@PathVariable long id) {
         return offerService.getById(id);
     }
 
     @PostMapping
-    @ResponseStatus(HttpStatus.CREATED)
-    public ResponseOfferDto addOffer(@RequestBody CreateOfferDto offerDto) {
+    public OfferDto addOffer(@RequestBody CreateOfferDto offerDto) {
         return offerService.addOffer(offerDto);
     }
+
+    @PutMapping("{id}")
+    public OfferDto updateOffer(@PathVariable long id, @RequestBody CreateOfferDto offerDto) {
+        return offerService.updateOffer(id, offerDto);
+    }
+
+    @DeleteMapping("{id}")
+    public OfferDto deleteOffer(@PathVariable long id) {
+        return offerService.deleteOffer(id);
+    }
+
+    @PostMapping("feature")
```

Comment:
To my understanding one Offer has many Features (one to many relationship) so - thinking RESTfully - a collection of Features is a subresource of Offer resource. So endpoint name (when adding new feature) should be:
```
POST /api/offers/{offerId}/features
```
instead of:
```
POST /api/offers/feature
```

Reply:
How about specifying particular method in url? In this example this would be `POST /api/offers/{offerId}/features/add` for me it makes endpoint more readable.

Reply:
I agree that verbs like ```/add```, ```/remove```, ```/read``` etc. in the URL might seem to increase readability and explicitness of your endpoints but they are incorrect when it comes to REST naming conventions.

Please notice that HTTP metods can be treated as verbs:
POST = add (to collection)
PUT = create/replace
GET - read
DELETE - remove
PATCH - update

So you don't need additional verbs in URL like ```/read```.